pandas/core/resample.py

```diff
@@ -853,13 +853,34 @@ def var(self, ddof=1, *args, **kwargs):
 
     @Appender(GroupBy.size.__doc__)
     def size(self):
-        # It's a special case as higher level does return
-        # a copy of 0-len objects. GH14962
         result = self._downsample("size")
-        if not len(self.ax) and isinstance(self._selected_obj, ABCDataFrame):
+        if not len(self.ax):
             from pandas import Series
 
-            result = Series([], index=result.index, dtype="int64")
+            if self._selected_obj.ndim == 1:
+                name = self._selected_obj.name
+                result = self._selected_obj.__class__(
+                    [], index=result.index, dtype="int64", name=name
+                )
+            else:
+                result = Series([], index=result.index, dtype="int64")
+        return result
+
+    @Appender(GroupBy.count.__doc__)
+    def count(self):
+        result = self._downsample("count")
+        if not len(self.ax):
+            if self._selected_obj.ndim == 1:
+                result = self._selected_obj.__class__(
+                    [], index=result.index, dtype="int64", name=self._selected_obj.name
+                )
+            else:
+                from pandas import DataFrame
+
+                result = DataFrame(
```

Comment:
this should also be self._selected_obj.__class__ (use a temporary variable here to avoid repeating)

Reply:
OK for `count()`.

> pls combine these; we don't need both paths; these are the same. if some tests need
> changing because the behavior changed, then pls do this.

This comment is now displayed as revolved. Was it an error?

As for merging branches in `size()`, I don't think they can be significantly merged because:

* in the `ndim==1` we should return `self._selected_obj.__class_` and pass the name;
* in `ndim>1`, we should generate a type with `ndim==1` and we don't have a name to pass.

The only way to merge code between these paths would be to do something like:

```pyhon
@Appender(GroupBy.size.__doc__)
def size(self):
    result = self._downsample("size")
    if not len(self.ax):        
        from pandas import Series
        if self._selected_obj.ndim == 1:
            name = self._selected_obj.name
            klass = self._selected_obj.__class__
            kwargs = dict(name=name)
        else:
            klass = Series
            kwargs = {}
        result = klass([], index=result.index, dtype="int64", **kwargs)
    return result
```

Which kind of behavior change would you expect?